omod/src/test/java/org/openmrs/web/attribute/handler/SerializingFieldGenDatatypeHandlerTest.java

```diff
@@ -41,4 +50,85 @@ public void getValue_shouldReturnTheCorrectTypedValue() throws Exception {
 		SerializingFieldGenDatatypeHandler handler = new MockLocationFieldGenDatatypeHandler();
 		Assert.assertEquals(expectedLocation, handler.getValue(datatype, request, formFieldName));
 	}
+
+	@Test
+	public void getValue_givenEmptyValue_shouldReturnNull() throws Exception {
+		final String locationUuid = "";
+		final String formFieldName = "uuid";
+		MockHttpServletRequest request = new MockHttpServletRequest();
+		request.setParameter(formFieldName, locationUuid);
+		MockLocationDatatype datatype = mock(MockLocationDatatype.class);
+		Location expectedLocation = mock(Location.class);
+		when(datatype.deserialize(eq(locationUuid))).thenReturn(expectedLocation);
+		SerializingFieldGenDatatypeHandler handler = new MockLocationFieldGenDatatypeHandler();
+		Assert.assertNull(handler.getValue(datatype, request, formFieldName));
+	}
+
+	@Test
+	public void getValue_givenLocationId_shouldCallDeserializeWithLocationUuid() throws Exception {
+		List<Location> locations = Context.getLocationService().getAllLocations();
+		Location testedLocation = locations.get(0);
```

Comment:
Instead of getting all and then the first, why not just get for a particular id? You can pick them from here: https://github.com/openmrs/openmrs-core/blob/master/api/src/test/resources/org/openmrs/include/standardTestDataset.xml

Reply:
@dkayiwa Done :)